config-model/src/main/java/com/yahoo/vespa/model/application/validation/first/AccessControlValidator.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.model.application.validation.first;
+
+import com.yahoo.config.model.deploy.DeployState;
+import com.yahoo.vespa.model.VespaModel;
+import com.yahoo.vespa.model.application.validation.Validator;
+import com.yahoo.vespa.model.container.ContainerCluster;
+import com.yahoo.vespa.model.container.component.Handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.yahoo.collections.CollectionUtil.mkString;
+import static com.yahoo.vespa.model.container.http.AccessControl.isBuiltinGetOnly;
+
+/**
+ * Validates that hosted applications in prod zones have write protection enabled.
+ *
+ * @author gjoranv
+ */
+public class AccessControlValidator extends Validator {
+
+    @Override
+    public void validate(VespaModel model, DeployState deployState) {
+
+        // TODO: what about zone application? (HOSTED_INFRASTRUCTURE)
+
+        if (deployState.isHosted() && deployState.zone().environment().isProduction()) {
+            List<String> offendingClusters = new ArrayList<>();
+            for (ContainerCluster cluster : model.getContainerClusters().values()) {
+                if (cluster.getHttp() == null
```

Comment:
When is this the case?

Reply:
When there is no 'http' element in services.xml. It's covered in the unit test `cluster_with_handler_fails_validation_without_http_element`